chord/chord.go

```diff
@@ -327,15 +340,15 @@ func runChord(fs *FileSystem, myIP string, myID uint64, port int, joinNode strin
 			} else {
 				log.Printf(green("Our successor: %v:%v"), fsRes.ret.SuccessorId, fsRes.ret.SuccessorIp)
 				chord.successor = fsRes.ret.SuccessorId
-				chord.finger[0] = chord.successor
+				chord.finger[0] = chord.successor // needed????
 				addToRing(fsRes.ret.SuccessorId, fsRes.ret.SuccessorIp, chord.ringMap)
 				notifyReq := &pb.NotifyArgs{PredecessorId: myID, PredecessorIp: myIP}
 				go chord.ringMap[chord.successor].conn.NotifyRPC(context.Background(), notifyReq)
 			}
 
 		// We received fix fingers response back from ourselves
 		case ff := <-chord.fixFingersResponseChan:
-			chord.finger[chord.next] = ff.ret.SuccessorId
+			chord.finger[ff.nextID] = ff.ret.SuccessorId
```

Comment:
allows finger table fixes to come in at any time. May be a risk if there are greater than M fixes in the channel, but I don't think it is that much of a risk